homeassistant/components/nobo_hub/climate.py

```diff
@@ -0,0 +1,208 @@
+"""Python Control of Nobø Hub - Nobø Energy Control."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from pynobo import nobo
+import voluptuous as vol
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    ATTR_TARGET_TEMP_HIGH,
+    ATTR_TARGET_TEMP_LOW,
+    PRESET_AWAY,
+    PRESET_COMFORT,
+    PRESET_ECO,
+    PRESET_NONE,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_MODE, ATTR_NAME, PRECISION_TENTHS, TEMP_CELSIUS
+from homeassistant.core import HomeAssistant, callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    ATTR_OVERRIDE_ALLOWED,
+    ATTR_TARGET_ID,
+    ATTR_TARGET_TYPE,
+    ATTR_TEMP_COMFORT_C,
+    ATTR_TEMP_ECO_C,
+    CONF_OVERRIDE_TYPE,
+    CONF_OVERRIDE_TYPE_NOW,
+    DOMAIN,
+)
+
+SUPPORT_FLAGS = (
+    ClimateEntityFeature.PRESET_MODE | ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+)
+
+PRESET_MODES = [PRESET_NONE, PRESET_COMFORT, PRESET_ECO, PRESET_AWAY]
+
+MIN_TEMPERATURE = 7
+MAX_TEMPERATURE = 40
+
+_LOGGER = logging.getLogger(__name__)
+
+_ZONE_NORMAL_WEEK_LIST_SCHEMA = vol.Schema({cv.string: cv.string})
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Nobø Ecohub platform from UI configuration."""
+
+    # Setup connection with hub
+    hub: nobo = hass.data[DOMAIN][config_entry.entry_id]
+
+    override_type = (
+        nobo.API.OVERRIDE_TYPE_NOW
+        if config_entry.options.get(CONF_OVERRIDE_TYPE) == CONF_OVERRIDE_TYPE_NOW
+        else nobo.API.OVERRIDE_TYPE_CONSTANT
+    )
+
+    # Add zones as entities
+    async_add_entities(
+        [NoboZone(zone_id, hub, override_type) for zone_id in hub.zones],
+        True,
+    )
+
+
+class NoboZone(ClimateEntity):
+    """Representation of a Nobø zone.
+
+    A Nobø zone consists of a group of physical devices that are
+    controlled as a unity.
+    """
+
+    _attr_max_temp = MAX_TEMPERATURE
+    _attr_min_temp = MIN_TEMPERATURE
+    _attr_precision = PRECISION_TENTHS
+    _attr_preset_modes = PRESET_MODES
+    # Need to poll to get preset change when in HVACMode.AUTO.
+    _attr_should_poll = True
+    _attr_supported_features = SUPPORT_FLAGS
+    _attr_temperature_unit = TEMP_CELSIUS
+
+    def __init__(self, zone_id, hub: nobo, override_type):
+        """Initialize the climate device."""
+        self._id = zone_id
+        self._nobo = hub
+        self._attr_unique_id = hub.hub_serial + ":" + zone_id
```

Comment:
Please use f-strings instead of string concatenation.

Reply:
Fixed.